include/felspar/io/completion.hpp

```diff
@@ -1,29 +1,52 @@
 #pragma once
 
-
 #include <felspar/coro/task.hpp>
+#include <felspar/io/exceptions.hpp>
 #include <felspar/test/source.hpp>
 
 #include <memory>
+#include <system_error>
 
 
 namespace felspar::io {
 
 
+    template<typename R>
+    class ec;
+    template<typename R>
+    class result;
     class warden;
 
 
+    namespace detail {
+        struct error_store {
+            felspar::source_location loc;
+            std::error_code code = {};
+            char const *message = "";
+            [[noreturn]] void throw_exception() {
```

Comment:
Should this function be `&&`-qualified?

Reply:
It could be, but TBH I think it probably makes more sense to remove the `std::move` around the `loc` member as the move brings zero benefit (all the internal fields have to be copied anyway): https://en.cppreference.com/w/cpp/header/source_location and my polyfill: https://github.com/Felspar/test/blob/main/include/felspar/test/source.hpp#L25-L48

Removing that would also mean a bit less syntax everywhere and the code being closer to the reality

Reply:
I've removed the `move`s